guardrails/run.py

```diff
@@ -757,9 +893,133 @@ async def async_validate(
 
             return validated_output
 
+    async def async_prepare(
+        self,
+        call_log: Call,
+        index: int,
+        instructions: Optional[Instructions],
+        prompt: Optional[Prompt],
+        msg_history: Optional[List[Dict]],
+        prompt_params: Dict,
+        api: Optional[Union[PromptCallableBase, AsyncPromptCallableBase]],
+        prompt_schema: Optional[StringSchema],
+        instructions_schema: Optional[StringSchema],
+        msg_history_schema: Optional[StringSchema],
+        output_schema: Schema,
+    ) -> Tuple[Optional[Instructions], Optional[Prompt], Optional[List[Dict]]]:
+        """Prepare by running pre-processing and input validation.
+
+        Returns:
+            The instructions, prompt, and message history.
+        """
+        with start_action(action_type="prepare", index=index) as action:
+            if api is None:
+                raise ValueError("API must be provided.")
+
+            if prompt_params is None:
+                prompt_params = {}
+
+            if msg_history:
```

Comment:
long control switch. Can we split some of the logic up?

For ex:
def start_message_history_action
def validate_msg_history_schema
def start_prompt_action
def validate_prompt_schema
def start_instruction_action
def validate_instruction_schema